FunctionalProgramming/Recursion/SuperDigit.fsx

```diff
@@ -1,37 +1,68 @@
-    input.Split ' '
+// I generally put such tiny functions on single lines; this saves space and
+// improves the reading flow, because a piece of code such as this function
+// body can get a bit "lost" visually.
+let splitBySpace (input : string) = input.Split ' '
 
-let readInputs _ =
-    let inputs = 
-        Console.ReadLine () 
-        |> splitBySpace 
-    inputs.[0], int64 inputs.[1]
+// While this works with `_`, the placeholder parameter of parameterless
+// functions is by convention always the `unit` value `()`. That makes it
+// immediately clear that there is no useful input to this function, while the
+// `_`  placeholder actually causes the parameter to become generic, which is
+// irritating, because while one can pass *anything*, it will never matter.
+let readInputs () =
+    // Destructuring assignments like this take a bit of getting used to,
+    // but are pretty powerful in that they let you name the individual values
+    // immediately without the need for an intermediate array and "anonymous"
+    // access via "magic number" indices.
+    // Pattern matching in this way incurs a compiler warning, because the
+    // match will fail with an exception for any array that doesn't have
+    // *exactly* two elements.
+    let [| n; k |] = Console.ReadLine () |> splitBySpace 
+    n, int64 k
 
-let calculateP (n, k:int64) =
-    [1L..k]
-    |> List.map(fun _ -> n)
-    |> String.Concat
+// I deleted `calculateP`, because it was not used.)
 
-let sumIndividualDigits (num:string) =
+// This isn't a case where you'd typically use "derivation" naming with an
+// apostrophe; this is really a separate function that might still make sense
+// outside of the function that it's being used from here.
+let sumDigits (num : string) =
     num.ToCharArray()
-    |> Seq.map(string >> int64)
-    |> Seq.sum
+    // Seq.sumBy is the composition of `Seq.map` and `Seq.sum` 
+    |> Seq.sumBy (string >> int64)
     |> string
 
-let sumIndividualDigits' (n, k:int64) =
-    k * (sumIndividualDigits n |> int64)
+// This is a bit less involved and doesn't require making sense of the whole
+// multiplication term at once.
+let sumIndividualDigits n (k : int64) =
+    n
+    |> sumDigits
+    |> int64
+    // Generally (there may be exceptions), operators can be turned into
+    // functions by putting them in parentheses. In this case, `(*)` behaves
+    // like a `multiply` function that takes two factors as parameters.
+    // This behavior can make for more pipe-friendly code as in this example,
+    // however, it should preferably only be used with commutative operations,
+    // because when the order of operands matters, this will read rather
+    // unintuitively.
+    |> (*) k
     |> string 
 
+// Pattern matching on booleans is a little odd; an `if` expression would be
+// more direct. Pattern matching on the actual value is the visually most
+// expressive representation of what this function does.
```

Comment:
Some people argue that `if` expressions should not be used at all in functional code, in favor of pattern matching. I generally do use them when the decision that's being made semantically really is a yes/no one, because in that case I feel it corresponds better with the mental model you have. Outside of that, pattern matching is better at saying "these are the possible values and what we're doing for each of them".